crates/project/src/lsp_store.rs

```diff
@@ -3527,6 +3530,12 @@ struct CodeLensData {
     update: Option<(Global, CodeLensTask)>,
 }
 
+#[derive(Debug, Default)]
+struct SemanticTokensData {
+    result_id: Option<String>,
```

Comment:
It's a bit odd to see this optional, we do know which server did we send the request to and got the response from.

Reply:
Yup, agreed. This was just so I could `#[derive(Default)]`. Now that `LspData` exists, it should be easy to get rid of this.